src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.cs

```diff
@@ -247,7 +253,7 @@ public override BoundNode VisitLambda(BoundLambda node)
             try
             {
                 _factory.CurrentFunction = node.Symbol;
-                return base.VisitLambda(node);
+                return base.VisitLambda(node)!;
```

Comment:
How do we know?

Reply:
Because the base visitor *never* returns null.  Only its subclasses do.  Asserting these would be very distracting and not accomplish much.

---
In reply to: [386592963](https://github.com/dotnet/roslyn/pull/42068#discussion_r386592963) [](ancestors = 386592963)